### PR TITLE
feat: Support setup.py files

### LINKED ISSUE #2681: Support python setup.py
https://packaging.python.org/discussions/install-requires-vs-requirements/#install-requires

### PR DESCRIPTION
**Pull request by kayoub5**
Sample PRs
https://github.com/kayoub5/talisker/pulls

Docker image `renovate/python` needs to be created with [mock](https://pypi.org/project/mock/) installed

Closes #2681

### CODE REVIEW

**Comment by rarkins** [comment]
@kayoub5 	can you submit a PR to https://github.com/renovatebot/docker-python for `mock`? I suppose best way for us is a `requirements.txt` with exact version that can itself be renovated?

File: lib/manager/pip_requirements/extract.js [hunk 1]
```diff
@@ -11,8 +11,23 @@ module.exports = {
   dependencyPattern,
   packagePattern,
   extractPackageFile,
+  isSkipComment,
```
Comment (rarkins) [inline lib/manager/pip_requirements/extract.js]: I'm wondering if this logic should logically exist independent of pip_requirements and instead be in some (new) python helper package

File: lib/manager/pip_setup/extract.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
+const { execSync } = require('child_process');
```
Comment (rarkins) [inline lib/manager/pip_setup/extract.js]: Do we want execSync or better to await a promisified async exec?

File: lib/manager/pip_requirements/extract.js (hunk 1 above)
Comment (kayoub5) [reply lib/manager/pip_requirements/extract.js]: Could we include this in all package managers?

Comment (rarkins) [reply lib/manager/pip_requirements/extract.js]: Yes. This syntax or something similar could be supported by additional package managers, including non-python

File: lib/manager/pip_setup/extract.js (hunk 2 above)
Comment (kayoub5) [reply lib/manager/pip_setup/extract.js]: `extractPackageFile` is not async, if it was refactored to be so, that would make this PR much easier.

Comment (rarkins) [reply lib/manager/pip_setup/extract.js]: We can easily add an `await` to where it's called, which would then allow each package manager to decide whether to make sync or async

**Pull request merged**